compiler_tester/run_tests.py

```diff
@@ -0,0 +1,303 @@
+                filename,
+                update_reference,
+                extra_args)
+        elif filename.endswith(".f"):
+            # Use fixed form
+            run_test(
+                filename,
+                "ast",
+                "lfortran --fixed-form --show-ast --no-color {infile} -o {outfile}",
+                filename,
+                update_reference,
+                extra_args)
+        else:
+            # Use free form
+            run_test(
+                filename,
+                "ast",
+                "lfortran --show-ast --no-color {infile} -o {outfile}",
+                filename,
+                update_reference,
+                extra_args)
+    if ast_indent:
+        run_test(
+            filename,
+            "ast_indent",
+            compiler + " --show-ast --indent --no-color {infile} -o {outfile}",
+            filename,
+            update_reference,
+            extra_args)
+
+    if ast_f90:
+        if filename.endswith(".f"):
+            # Use fixed form
+            run_test(
+                filename,
+                "ast_f90",
+                "lfortran --fixed-form --show-ast-f90 --no-color {infile}",
+                filename,
+                update_reference,
+                extra_args)
+        else:
+            # Use free form
+            run_test(
+                filename,
+                "ast_f90",
+                "lfortran --show-ast-f90 --no-color {infile}",
+                filename,
+                update_reference,
+                extra_args)
+
+    if ast_openmp:
+        run_test(
+            filename,
+            "ast_openmp",
+            "cpptranslate --show-ast-openmp {infile}",
+            filename,
+            update_reference)
+
+    if asr:
+        run_test(
+            filename,
+            "asr",
+            compiler + " --show-asr --no-color {infile} -o {outfile}",
+            filename,
+            update_reference,
+            extra_args)
+
+    if asr_implicit_typing:
+        run_test(
+            filename,
+            "asr",
+            "lfortran --show-asr --implicit-typing --no-color {infile} -o {outfile}",
+            filename,
+            update_reference,
+            extra_args)
+
+    if asr_preprocess:
+        run_test(
+            filename,
+            "asr_preprocess",
+            compiler + " --cpp --show-asr --no-color {infile} -o {outfile}",
+            filename,
+            update_reference,
+            extra_args)
+
+    if asr_indent:
+        run_test(
+            filename,
+            "asr_indent",
+            compiler + " --show-asr --indent --no-color {infile} -o {outfile}",
+            filename,
+            update_reference,
+            extra_args)
+
+    if mod_to_asr:
+        run_test(
+            filename,
+            "mod_to_asr",
+            "lfortran mod --show-asr --no-color {infile}",
+            filename,
+            update_reference)
```

Comment:
I am worried that once we have 10 frontends, we'll have to maintain the union of these.

I think the `compiler_tester` should contain code that is used by all frontends, the "common" code. The above does not look like common code.

If I have time, I'll try to help with this to get this done.